src/network/networkd-dhcp4.c

```diff
@@ -610,7 +610,10 @@ int dhcp4_check_promote_secondaries(Link *link) {
         assert(link->network->dhcp & ADDRESS_FAMILY_IPV4);
 
         if (!(check_promote_secondaries("all") ||check_promote_secondaries("default") || check_promote_secondaries(link->ifname))) {
-                log_warning("net.ipv4.conf.%s.promote_secondaries is unset, DHCP may misbehave if IP changes between leases", link->ifname);
+                _cleanup_free_ char *promote_secondaries_path = NULL;
+
+                promote_secondaries_path = strjoin("net/ipv4/conf/", link->ifname, "/promote_secondaries");
+                return sysctl_write(promote_secondaries_path, "1");
```

Comment:
Please make this configurable we should not be forcing for it. And I think it should be defaults to true. A boolen should be configured via `PromoteSecondaries=`  and at `.network`  file.